### DISCUSSION TITLE
can't catch telegram API exception

### DISCUSSION THREAD

**Discussion by mifizz** [eternnoir/pyTelegramBotAPI#2410] [Q&A] [answered]
I am using **polling()** in a **while** loop, so I can make my own logging and exception handling system. It catches all exceptions and handles them except for telegram API exception. If telegram API exception occurs, my bot just crashes, because **try... except...** does not catch this exception. How am I supposed to catch telegram API exception?

**Comment by Badiboy** [comment, marked as answer]
Pass your exception handler to Telebot (exception_handler...) to catch whatever you want.
